ramda.js

```diff
@@ -4705,6 +4707,42 @@
     R.gte = op(function _gte(a, b) { return a >= b; });
 
 
+    function createMaxMin(comparator, initialVal) {
+        return function(list) {
+            if (arguments.length === 0) {
+                throw noArgsException();
+            }
+            var idx = -1, winner = initialVal, computed;
+            while (++idx < list.length) {
+                computed = +list[idx];
+                if (comparator(computed, winner)) {
+                    winner = computed;
+                }
+            }
+            return winner;
+        };
+    }
+
+    function createMaxMinWith(comparator) {
+        return function(valueComputer, list) {
+            if (!(list && list.length > 0)) {
+                return;
```

Comment:
Note that `maxBy` does not share that property.  It's quite possible that 

    maxBy(myFunc, [a, b, ...]) != maxBy(myFunc, [b, a, ...])

And this doesn't bother me at all.

Our algorithm is functionally equivalent to your suggestion.  The problem is that `max` and `maxBy` have two fundamentally different concerns.  One has to do with values, the other with structures which can -- just for comparison purposes -- be represented by a value.  If we wanted to return that value for `maxBy`, we'd have no issue, but we want to return the original structure.

    var grades = [{ltr: 'B+', nbr: 87}, {ltr: 'B+', nbr: 89}, {ltr: 'C', nbr: 74}];
    maxBy(prop('nbr'), grades); //=> {ltr: 'B+', nbr: 89}

I think any of Ramda's `xxxBy` functions can be thought generally of projecting their data before processing it, but then, of course, restoring it before returning it to the user.  Since most projections lose information, there is no issue when things that otherwise wouldn't be equal end up equal.  That's just part of the process.

That there's some projection in `max` (because of the coercion) would be more troubling, but it comes out in the wash, since we end up returning the same numeric value regardless of which original value was coerced into it.